Rn 2.41.0 (#1158)
percona/pmm-doc: docs/release-notes/2.41.0.md

# Percona Monitoring and Management 2.41.0

| **Release date:** | Oct 05, 2023                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

<!---

!!! caution alert alert-warning "Important/Caution"
    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

--->

## Release Highlights

> 

Reviewer: ```suggestion
### Data collection is not can survive temporary Network problems between Database and PMM server 
The PMM software is designed to monitor and troubleshoot, but there may be instances where it is unable to collect data due to connection issues. This is usually caused by a disruption in communication between the PMM server and the monitored databases. In the past, this resulted in a period of time where no data was collected. However, with the latest release, all types of data, including Service Metrics and Query Metrics, will be collected during such disruptions. This data will be stored and visible on the UI as soon as the connection is restored. This will enable users to investigate issues caused by server load or other problems that may have disrupted the data collection.

### Simplified Reporting about Database problems to Percona
Efficient resolution of a problem often relies on a well-written report detailing the issue. 
To simplify the reporting process for our customers, we created an internal tool some time ago. We later added it to the PMM server, but its execution was complex, requiring users to install additional UI or use the terminal. 
However, in this release, we have made the tool PMM Dump(https://docs.percona.com/pmm-dump-documentation/index.html) easily accessible for all PMM users through the PMM user interface. 
Both Percona customers and community users can benefit from its use in collecting data for bug reports or internal usage. This marks the first step towards improving data gathering and triaging problems more efficiently.
 
### PostgreSQL monitoring performance improvement
At our company, performance is a crucial aspect of our work and we prioritize the principle of "Do no Harm." Recently, we received reports about excessive resource usage in monitoring PostgreSQL, particularly with regard to connections being used. In this latest release, we have made further improvements to optimize resource usage for monitoring, resulting in pmm using significantly fewer connections compared to previous versions and other resources.

### DBaaS deprecation, Long live to Everest
We have achieved positive outcomes in developing our Database as a Service in PMM. As we move forward to the next level, we have decided to separate the DBaaS functionality into an independent product. Consequently, we are discontinuing the DBaaS functionality in PMM and offering migration guides to Everest[LINK].

The functionality will still be available in PMM v2, but there will be no further updates. We strongly advise users to utilize Everest for their Database deployments in K8s.

```

# Percona Monitoring and Management 2.41.0

| **Release date:** | Oct 05, 2023                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

<!---

!!! caution alert alert-warning "Important/Caution"
    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

--->

## Release Highlights

> ### Data collection is not can survive temporary Network problems between Database and PMM server 

Reviewer 2: ```suggestion
### Data collection is now can survive temporary Network problems between Database and PMM server 
```

> The PMM software is designed to monitor and troubleshoot, but there may be instances where it is unable to collect data due to connection issues. This is usually caused by a disruption in communication between the PMM server and the monitored databases. In the past, this resulted in a period of time where no data was collected. However, with the latest release, all types of data, including Service Metrics and Query Metrics, will be collected during such disruptions. This data will be stored and visible on the UI as soon as the connection is restored. This will enable users to investigate issues caused by server load or other problems that may have disrupted the data collection.

Reviewer: ```suggestion
PMM data is essential for monitoring and troubleshooting by its users. The PMM application operates through a client-server system, exchanging data between clients (on DB servers) and servers (PMM UI interface and data storage). Previously, when network or application issues interrupted this connection, PMM could not collect complete data. However, with this latest release, such problems with the PMM Client and Server components will no longer result in data loss. Instead, data gathered during the connection problem will be delivered immediately once the connection is re-established. This new feature will allow PMM users to investigate problems that previously could not be examined, providing a clearer understanding of database or network problems.
```

### Simplified Reporting about Database problems to Percona

> Efficient resolution of a problem often relies on a well-written report detailing the issue. 
> To simplify the reporting process for our customers, we created an internal tool some time ago. We later added it to the PMM server, but its execution was complex, requiring users to install additional UI or use the terminal. 
> However, in this release, we have made the tool PMM Dump(https://docs.percona.com/pmm-dump-documentation/index.html) easily accessible for all PMM users through the PMM user interface. 
> Both Percona customers and community users can benefit from its use in collecting data for bug reports or internal usage. This marks the first step towards improving data gathering and triaging problems more efficiently.

Reviewer: ```suggestion
Efficient resolution of a problem often relies on a well-written report detailing the issue. 
We have added a new feature that allows both Users and Customers to gather data from PMM. This data can be shared with Percona support to help solve any issues with Customer Databases in SLA or assist our Support team in addressing community-reported bugs in Jira.
To simplify the reporting process for our customers, we created an internal tool some time ago. We later added it to the PMM server, but its execution was complex, requiring users to install additional UI or use the terminal. 
However, in this release, we have made the tool PMM Dump(https://docs.percona.com/pmm-dump-documentation/index.html) easily accessible for all PMM users through the PMM user interface. 
```

 
### PostgreSQL monitoring performance improvement

> At our company, performance is a crucial aspect of our work and we prioritize the principle of "Do no Harm." Recently, we received reports about excessive resource usage in monitoring PostgreSQL, particularly with regard to connections being used. In this latest release, we have made further improvements to optimize resource usage for monitoring, resulting in pmm using significantly fewer connections compared to previous versions and other resources.

Reviewer: ```suggestion
At our company, we prioritize the principle of "Do no Harm" and recognize that performance is a crucial aspect of our work. With this release, we have addressed a common problem for our PostgreSQL users regarding efficient resource usage by PMM. Specifically, we have decreased the number of used connections for monitoring, resulting in improved efficiency.
```

Reviewer: moved doc to https://www.notion.so/percona/PMM-2-41-3015eb1f559242008595906b117dff5f

# Percona Monitoring and Management 2.41.0

| **Release date:** | Nov 25, 2023                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

<!---

!!! caution alert alert-warning "Important/Caution"
    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

--->

## Release Highlights

### Continuous data collection during outages

> PMM data is crucial for monitoring and troubleshooting. PMM operates on a client-server architecture, exchanging data between clients (on database servers) and servers (PMM UI interface and data storage). If the PMM connection was interrupted by network or application issues in the previous versions of PMM, the system could not collect complete data, resulting in data loss. This issue has been addressed in PMM 2.41.0. If the PMM client and server components encounter connection problems, the data gathered during these periods is stored and transmitted immediately after the connection is restored.

Reviewer 3: I think it means that previously we some data was lost during network outages so after connection was restored they ended up with incomplete data

PMM users can now investigate challenging and complex problems, thus providing a deeper understanding of database and network issues.

### Streamlined database problem reporting to Percona

> Effective database problem resolution relies on the quality and comprehensiveness of issue reporting.

Reviewer 3: agreed

> To improve the process of gathering and sharing PMM metrics and data, we've now integrated the [pmm_dump client utility](https://docs.percona.com/pmm-dump-documentation/index.html) into PMM. Originally a standalone client for PMM Server, this now readily accessible within the PMM user interface.

Reviewer 3: Because the utility app works differently and the instructions are relevant only for users who need to work with it outside PMM.

This integration enables you to effortlessly collect data from PMM and empowers our Support team to better assist customers in solving database issues in SLA and address community-reported bugs in Jira.

To get started, go to the **Help** section on the main menu > **PMM Dump** and choose to either export a dataset locally or upload it via an SFTP server using the credentials generated through your Percona Support ticket.

### PostgreSQL monitoring performance improvement

In our continuing efforts to improve PMM, we recognize the significance of performance and its impact on our users. Starting with PMM 2.41.0, we have optimized resource usage for PostgreSQL users. This has been accomplished by reducing the number of connections used in monitoring.

### PMM DBaaS functionality evolution into Percona Everest

We have achieved positive outcomes in developing PMM's Database-as-a-Service feature. As we move to the next level, we have decided to separate the DBaaS functionality into an independent product. Consequently, we are discontinuing the DBaaS functionality in PMM and offering a [migration path to Everest](http://per.co.na/pmm-to-everest-guide).

> While the DBaaS functionality will remain available in PMM v2, all future updates and enhancements will be exclusively accessible through the Percona Everest interface.

Reviewer 4: Is DBaaS available in all 2.x versions of PMM?

If not, let's rephrase the text:
... available in all PMM versions 2.x and later, all ...

Reviewer 3: @rnovikovP, just to confirm, DBaaS current functionality will be available in PMM 2.x **and later**, right?

Reviewer: DBaaS functionality will be present as "deprecated" in all future pmm v2 releases (patch releases only planned for now). 
For the v3, we are completely removing it, and this is one of the main changes in the v3 scope.

Reviewer 3: ok, removed "and later" bit.

> We strongly recommend you try [Percona Everest](http://per.co.na/pmm-to-everest) for a more streamlined and robust experience with database deployments.

Reviewer 4: Let's remove the adverb.

Phrasing suggestion:
We strongly recommend you try [Percona Everest](http://per.co.na/pmm-to-everest) for a more streamlined and robust experience with database deployments.

to

We strongly recommend you try [Percona Everest](http://per.co.na/pmm-to-everest) for a more streamlined and robust experience with database deployments.

Always start with the "For":
For a more streamlined and robust database deployment experience, try [Percona Everest](http://per.co.na/pmm-to-everest).

## New Features

> [PMM-8655](https://jira.percona.com/browse/PMM-8655) - Alongside Victoria metrics, PMM now also captures QAN data during network outages. Once the connection is restored, all accumulated data is promptly delivered, ensuring you have comprehensive information to work with across your instances.

Reviewer 4: Phrasing suggestion:
Alongside Victoria metrics, PMM now also captures QAN data during network outages. Once the connection is restored, all accumulated data is promptly delivered, ensuring you have comprehensive information to work with across your instances.

to

PMM now captures both QAN and Victoria metrics data during network outages and delivers it as soon as the connection is restored.

Where does it deliver the data, BTW?

"...ensuring you have comprehensive information to work with across your instances." - this is fluff

> [PMM-12459](https://jira.percona.com/browse/PMM-12459) - The [pmm_dump client utility](https://docs.percona.com/pmm-dump-documentation/index.html) previously available as a standalone client for PMM Server is now readily accessible within the PMM user interface. 

Reviewer 4: Are all these items supposed to start on different rows?

They are all bundled together in the same paragraph for me.

Phrasing suggestion here:
The [pmm_dump client utility] previously available as a standalone client for PMM Server is now readily accessible within the PMM user interface.

to 

You can now access the [pmm_dump client utility]), previously available as a standalone client for PMM Server, in the PMM user interface.

> [PMM-12463](https://jira.percona.com/browse/PMM-12463) - [Telemetry] Collecting information about env variables usage in PMM

Reviewer 4: Every time you're collecting something from me, it makes me nervous. Why don't you tell me exactly what you're collecting?

Also, let's not shorten "environment".

Another tjhing, please keep things in active voice:
Percona/we now collect(s) information on environment variable usage in PMM.

Reviewer: We have a doc about in-app functionality to see what's collected. 
We are not planning to keep track in RN as this is available from the pmm itself and can be changed often.

## Improvements

> [PMM-11341](https://jira.percona.com/browse/PMM-11341) - PMM 2.41.0 introduces limit for Auto-discovery in PostgreSQL, a feature that dynamically discovers all databases in your PostgreSQL instance. Limiting Auto-discovery reduces connections and prevents high CPU and RAM usage caused by multiple databases.

Reviewer 4: I'm not sure what you're recommending here. Should I limit auto-discovery?

Seems like if I don't, I may have a problem.

Reviewer 5: @alexfiru Without limit performance should stay same like before. If you want better performance then you can try set limit. In ideal case lower than 10.

[PMM-12375](https://jira.percona.com/browse/PMM-12375) - Starting with PMM 2.41.0, each instance of a service gets a `version` attribute to the response of the endpoint that provides a list of services being monitored by PMM.

> [PMM-12422](https://jira.percona.com/browse/PMM-12422) - Starting with PMM 2.41.0, a new flag called `--expose-exporter-address` has been introduced. When you enable this flag and set the metrics mode to **push**, the node exporter for that agent will use the IP address `0.0.0.0`. This means that any IP address, either from a local system or from anywhere on the internet, can access it. If the flag is not enabled, the node exporter will use the address `127.0.0.1`, the default name for the computer you are working on, or the localhost.

Reviewer 4: This is passive voice. Let's turn it to active. Let's keep consistency here. The previous item uses active voice.

> [PMM-12544](https://jira.percona.com/browse/PMM-12544) - Added deprecation notice to the DBaaS pages, to advise about the evolution of PMM DBaaS functionality into [Percona Everest](http://per.co.na/pmm-to-everest).

Reviewer 4: Phrasing sugg:
Added deprecation notice to the DBaaS pages, to advise about the evolution of PMM DBaaS functionality into [Percona Everest](http://per.co.na/pmm-to-everest). 

to

Added deprecation notices to the PMM documentation DBaaS pages. For a more streamlined and robust database deployment experience, try [Percona Everest](http://per.co.na/pmm-to-everest).

> [PMM-12549](https://jira.percona.com/browse/PMM-12549) - Added support for the new MongoDB version, which means you can use PMM to monitor MongoDB 7 databases.

Reviewer 4: Phrasing sugg:
Added support for the new MongoDB version, which means you can use PMM to monitor MongoDB 7 databases.

to

Added support for the latest MongoDB version. You can now use PMM to monitor MongoDB 7 databases.

## Components upgrade

> - [PMM-12154](https://jira.percona.com/browse/PMM-12154) - postgres_exporter has been updated to version [0.14.0](https://github.com/prometheus-community/postgres_exporter/releases).

Reviewer 4: Passive voice. The other item uses active voice. Let's keep active voice consistency here.

- [PMM-12223](https://jira.percona.com/browse/PMM-12223) - Updated Clickhouse to v22.8.

## Bugs Fixed

> - [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports were missing. As a result, you can investigate out-of-memory issues more efficiently and precisely.

Reviewer 4: The final sentence doesn't add anything of value here.
You're writing as if this is an enhancement. It's not. We're not giving users anything new that enables them to investigate out-of-memory issues more efficiently and precisely.

> - [PMM-12576](https://jira.percona.com/browse/PMM-12576) - After upgrading to PMM 2.40.0, changing the Admin user’s password from the terminal was not functioning. The issue has been resolved now.

Reviewer 4: Phrasing sugg:
After upgrading to PMM 2.40.0, changing the Admin user’s password from the terminal was not functioning. The issue has been resolved now.

to

After upgrading to PMM 2.40.0, changing the Admin user’s password using the terminal did not work. The issue has been resolved.

> - [PMM-12626](https://jira.percona.com/browse/PMM-12626) - The GO version that we used to build PMM was updated to the latest version to mitigate potential vulnerabilities.

Reviewer 4: Phrasing sugg:
The GO version that we used to build PMM was updated to the latest version to mitigate potential vulnerabilities.

to 

We updated the GO version used to build PMM to the latest version, to mitigate potential vulnerabilities.

Keep active voice.

To get started, go to the **Help** section on the main menu > **PMM Dump** and choose to either export a dataset locally or upload it via an SFTP server using the credentials generated through your Percona Support ticket.

### PostgreSQL monitoring performance improvement

In our continuing efforts to improve PMM, we recognize the significance of performance and its impact on our users. Starting with PMM 2.41.0, we have optimized resource usage for PostgreSQL users. This has been accomplished by reducing the number of connections used in monitoring.

### PMM DBaaS functionality evolution into Percona Everest

We have achieved positive outcomes in developing PMM's Database-as-a-Service feature. As we move to the next level, we have decided to separate the DBaaS functionality into an independent product. Consequently, we are discontinuing the DBaaS functionality in PMM and offering a [migration path to Everest](http://per.co.na/pmm-to-everest-guide).

While the DBaaS functionality will remain available in PMM v2, all future updates and enhancements will be exclusively accessible through the Percona Everest interface.

> We strongly recommend you try [Percona Everest](http://per.co.na/pmm-to-everest) for a more streamlined and robust experience with database deployments.

Reviewer 2: should the link be `pmm-to-everest`?

## New Features

[PMM-8655](https://jira.percona.com/browse/PMM-8655) - Alongside Victoria metrics, PMM now also captures QAN data during network outages. Once the connection is restored, all accumulated data is promptly delivered, ensuring you have comprehensive information to work with across your instances.
[PMM-12459](https://jira.percona.com/browse/PMM-12459) - The [pmm_dump client utility](https://docs.percona.com/pmm-dump-documentation/index.html) previously available as a standalone client for PMM Server is now readily accessible within the PMM user interface. 
[PMM-12463](https://jira.percona.com/browse/PMM-12463) - [Telemetry] Collecting information about env variables usage in PMM

## Improvements

[PMM-11341](https://jira.percona.com/browse/PMM-11341) - PMM 2.41.0 introduces limit for Auto-discovery in PostgreSQL, a feature that dynamically discovers all databases in your PostgreSQL instance. Limiting Auto-discovery reduces connections and prevents high CPU and RAM usage caused by multiple databases.

> [PMM-12375](https://jira.percona.com/browse/PMM-12375) - Starting with PMM 2.41.0, each instance of a service gets a `version` attribute to the response of the endpoint that provides a list of services being monitored by PMM.

Reviewer 2: I would highlight PMM UI on this item.
```suggestion
[PMM-12375](https://jira.percona.com/browse/PMM-12375) - Starting with PMM 2.41.0, each instance of a service gets a `version` attribute in the PMM Inventory UI.
```

> [PMM-12422](https://jira.percona.com/browse/PMM-12422) - Starting with PMM 2.41.0, a new flag called `--expose-exporter-address` has been introduced. When you enable this flag and set the metrics mode to **push**, the node exporter for that agent will use the IP address `0.0.0.0`. This means that any IP address, either from a local system or from anywhere on the internet, can access it. If the flag is not enabled, the node exporter will use the address `127.0.0.1`, the default name for the computer you are working on, or the localhost.

Reviewer 2: ```suggestion
[PMM-12422](https://jira.percona.com/browse/PMM-12422) - Starting with PMM 2.41.0, a new flag called `--expose-exporter-address` has been introduced. When you enable this flag any IP address, either from a local system or from anywhere on the internet, can access exporter endpoints. If the flag is not enabled, the exporter will be available only locally.
```
I think we can remove technical details about 0.0.0.0 here and just leave the part about being available from anywhere and localhost.

[PMM-12544](https://jira.percona.com/browse/PMM-12544) - Added deprecation notice to the DBaaS pages, to advise about the evolution of PMM DBaaS functionality into [Percona Everest](http://per.co.na/pmm-to-everest).

> [PMM-12549](https://jira.percona.com/browse/PMM-12549) - PMM 2.41.0 now supports MongoDB v7.

Reviewer: i think we need some clarification here - what this means exactly, to avoid false expectations in new features

Author: @BupycHuk could you please provide your inputs here.

Author: Done.

Reviewer: There was even internal confusion about what "support" means. 
In our case - it's purely "can monitor," but no new cool fancy features highlighted.

## Components upgrade

> - [PMM-12154](https://jira.percona.com/browse/PMM-12154) - postgres_exporter has been updated to version [0.14.0](https://github.com/prometheus-community/postgres_exporter/releases).

Reviewer: some basic info needed - "why  we did this " what is the user impact for this

Author: @BupycHuk Please provide this information.

Author: Done.

> - [PMM-12223](https://jira.percona.com/browse/PMM-12223) - Updated Clickhouse to v22.8.

Reviewer: Same here. some more generic phrases, maybe about performance/security if any

Author: @BupycHuk Please provide your inputs here.

Author: @BupycHuk Please provide info.

Reviewer 2: everythings below [this](https://clickhouse.com/docs/en/whats-new/changelog#-clickhouse-release-238-lts-2023-08-31)
everything in this [page](https://clickhouse.com/docs/en/whats-new/changelog/2022)
and everything above [this](https://clickhouse.com/docs/en/whats-new/changelog/2021#clickhouse-release-213-lts). So let's just write in common words. That it was done for better stability, performance and security.

Author: Done.

Reviewer 2: @rasika-chivate the version is still incorrect

## Bugs Fixed

> - [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports were missing. As a result, you can investigate out-of-memory issues more efficiently and precisely.

Reviewer 2: ```suggestion
- [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports for postgres_exporter were missing. As a result, you can investigate out-of-memory issues more efficiently and precisely.
```

> - [PMM-4712](https://jira.percona.com/browse/PMM-4712) - PMM frequently crashed due to out-of-memory issues caused by postgres_exporter consuming significant RAM. We could not investigate the issue as the pprof heap reports were missing. The pprof heap reports are now available.

Reviewer: sounds like we fixed the OOM issue but later, it's - about pprof issue.
Maybe start from "fixed absence of pprof  , so now we can investigate oom better "

Author: Fixed.

Author: Done.

> - [PMM-12576](https://jira.percona.com/browse/PMM-12576) - After upgrading to PMM 2.40.0, changing the Admin user’s password from the terminal was not functioning. The issue has been resolved now.

Reviewer 2: it was fixed in 2.40.1, should we mention it again?

Author: We can remove

> - [PMM-12626](https://jira.percona.com/browse/PMM-12626) - We have upgraded to the latest version of GO in order to prevent any potential vulnerabilities.

Reviewer: maybe something like "version go we used to build pmm" as we are not providing GO directly to the users

Author: Updated.

- [PMM-12549](https://jira.percona.com/browse/PMM-12549) - Added support for the latest MongoDB version. You can now use PMM to monitor MongoDB 7 databases.

## Components upgrade

- [PMM-12154](https://jira.percona.com/browse/PMM-12154) - Updated postgres_exporter to version [0.14.0](https://github.com/prometheus-community/postgres_exporter/releases). With this update, we have resolved several performance issues and eliminated the creation of multiple connections per database.
- [PMM-12223](https://jira.percona.com/browse/PMM-12223) - Clickhouse has been updated to version 23.8.2.7, which optimizes memory and CPU usage to improve system performance.

## Bugs Fixed

- [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports for postgres_exporter were missing. 
- [PMM-12626](https://jira.percona.com/browse/PMM-12626) - We updated the GO version used to build PMM to its latest version to mitigate potential vulnerabilities.
- [PMM-12658](https://jira.percona.com/browse/PMM-12658) - Corrected a typo in the MongoDB cluster summary dashboard.

> 

Reviewer: ```suggestion
- PMM-11714 - must in RN. We have some ADDITIONAL RELEASE NOTES TEXT for it . changing the LEvel - to public
- PMM-12414, PMM-12631
- PMM-12562 - need to be rechecked with QA maybe it's a defect for current version 
- PMM-12576
- PMM-12630
- PMM-12660

```

Reviewer: based on versions, it's not a defect, as 2.37.1 were affected, not 2.41 
<img width="977" alt="image" src="https://github.com/percona/pmm-doc/assets/25495422/14e9c975-70ed-4128-8650-6c1cee85ca92">

Author: Added.

- [PMM-12549](https://jira.percona.com/browse/PMM-12549) - Added support for the latest MongoDB version. You can now use PMM to monitor MongoDB 7 databases.

## Components upgrade

- [PMM-12154](https://jira.percona.com/browse/PMM-12154) - Updated postgres_exporter to version [0.14.0](https://github.com/prometheus-community/postgres_exporter/releases). With this update, we have resolved several performance issues and eliminated the creation of multiple connections per database.
- [PMM-12223](https://jira.percona.com/browse/PMM-12223) - Clickhouse has been updated to version 23.8.2.7, which optimizes memory and CPU usage to improve system performance.

## Bugs Fixed

- [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports for postgres_exporter were missing. 
- [PMM-12626](https://jira.percona.com/browse/PMM-12626) - Due to the packages being built on an outdated Go version, there was a potential vulnerability. We have updated Go to the latest version to mitigate this risk.
- [PMM-12658](https://jira.percona.com/browse/PMM-12658) - Corrected a typo in the MongoDB cluster summary dashboard.

> - [PMM-12414](https://jira.percona.com/browse/PMM-12414) - When the Alertmanager configuration included the `group_by` parameter, accessing the `logs.zip` endpoint resulted in an unexpected behavior (502 response).

Reviewer: ```suggestion
- [PMM-12414](https://jira.percona.com/browse/PMM-12414) - There was an unexpected behavior (502 response) when accessing the `logs.zip` endpoint. This was caused by the `group_by` parameter being included in the Alertmanager configuration.
```

to start with a problem we fixed, not the reason of its existence

- [PMM-12725](https://jira.percona.com/browse/PMM-12725) - Fixed the pagination for QAN.
- [PMM-11714](https://jira.percona.com/browse/PMM-11714) - Registering a node with the Grafana Admin flag enabled but a non-admin role was failing. This issue has now been resolved.
- [PMM-12660](https://jira.percona.com/browse/PMM-12660) - Prior to version 2.41.0 of PMM, the endpoint `/v1/management/Agent/List` could deliver database certificates to the PMM UI, allowing an authenticated admin user to view the output of TLS certificates. This posed a security issue since certificates should be consumed by the backend only. We have resolved this issue now.
- [PMM-12630](https://jira.percona.com/browse/PMM-12630) - Fixed an issue in which the upgrade process for PMM versions lower than or equal to 2.37.1 was stuck in a loop and could not be completed.

## Known issue

> - [PMM-12682](https://jira.percona.com/browse/PMM-12682) PMM doesn't show service names within the PMM Dump dataset when the **All Services** option is selected during PMM Dump creation. This issue will be fixed in a future release.

Reviewer: Status: MERGED
Probably no longer the bug  / defect

Author: @catalinaadam I think we can remove this.

- [PMM-12223](https://jira.percona.com/browse/PMM-12223) - Clickhouse has been updated to version 23.8.2.7, which optimizes memory and CPU usage to improve system performance.

## Bugs Fixed

- [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports for postgres_exporter were missing. 
- [PMM-12626](https://jira.percona.com/browse/PMM-12626) - Due to the packages being built on an outdated GO version, there was a potential vulnerability. We have updated GO to the latest version to mitigate this risk.
- [PMM-12658](https://jira.percona.com/browse/PMM-12658) - Corrected a typo in the MongoDB cluster summary dashboard.
- [PMM-12414](https://jira.percona.com/browse/PMM-12414) - When the Alertmanager configuration included the `group_by` parameter, accessing the `logs.zip` endpoint resulted in an unexpected behavior (502 response).
- [PMM-12631](https://jira.percona.com/browse/PMM-12631) - 
- [PMM-12725](https://jira.percona.com/browse/PMM-12725) - Fixed the pagination for QAN.

> - [PMM-11714](https://jira.percona.com/browse/PMM-11714) - During the upgrade from PMM 2.34.0 to 2.35.0, node metrics were not captured when registering a node with the Grafana Admin flag enabled but a non-admin role. This issue has now been resolved.

Reviewer 2: ```suggestion
- [PMM-11714](https://jira.percona.com/browse/PMM-11714) - Registering a node with the Grafana Admin flag enabled but a non-admin role was failing. This issue has now been resolved.
```

- [PMM-12544](https://jira.percona.com/browse/PMM-12544) - Added deprecation notices to the PMM documentation DBaaS pages. For a more streamlined and robust database deployment experience, try [Percona Everest](http://per.co.na/pmm-to-everest).
- [PMM-12549](https://jira.percona.com/browse/PMM-12549) - Added support for the latest MongoDB version. You can now use PMM to monitor MongoDB 7 databases.

## Components upgrade

- [PMM-12154](https://jira.percona.com/browse/PMM-12154) - Updated postgres_exporter to version [0.14.0](https://github.com/prometheus-community/postgres_exporter/releases). With this update, we have resolved several performance issues and eliminated the creation of multiple connections per database.
- [PMM-12223](https://jira.percona.com/browse/PMM-12223) - Clickhouse has been updated to version 23.8.2.7, which optimizes memory and CPU usage to improve system performance.

## Bugs Fixed

- [PMM-4712](https://jira.percona.com/browse/PMM-4712) - We have addressed the issue where the [pprof](https://github.com/google/pprof) heap reports for postgres_exporter were missing. 

> - [PMM-12626](https://jira.percona.com/browse/PMM-12626) - Due to the packages being built on an outdated GO version, there was a potential vulnerability. We have updated GO to the latest version to mitigate this risk.

Reviewer 2: seems ok

- [PMM-12658](https://jira.percona.com/browse/PMM-12658) - Corrected a typo in the MongoDB cluster summary dashboard.

> - [PMM-12414](https://jira.percona.com/browse/PMM-12414) - When the Alertmanager configuration included the group_by parameter, accessing the logs.zip endpoint resulted in an unexpected behavior (502 response).

Reviewer 2: We also removed alertmanager related files from logs.zip since we deprecated usage of alertmanager, should we include that?

> - [PMM-12631](https://jira.percona.com/browse/PMM-12631) - 

Reviewer 2: it's a duplicate of the task before it, so they can be merged or one of them can be removed.

> - [PMM-12631](https://jira.percona.com/browse/PMM-12631) - 

Author: Will remove it.

> - [PMM-12626](https://jira.percona.com/browse/PMM-12626) - Due to the packages being built on an outdated GO version, there was a potential vulnerability. We have updated the GO to the latest version to mitigate this risk.

Reviewer 2: looks ok

- [PMM-12682](https://jira.percona.com/browse/PMM-12682) - 
- [PMM-12683](https://jira.percona.com/browse/PMM-12683) - 
- [PMM-12725](https://jira.percona.com/browse/PMM-12725) - Fixed the pagination for QAN.
- [PMM-11714](https://jira.percona.com/browse/PMM-11714) - During the upgrade from PMM 2.34.0 to 2.35.0, node metrics were not captured when registering a node with the Grafana Admin flag enabled but a non-admin role. This issue has now been resolved.

> - [PMM-12562](https://jira.percona.com/browse/PMM-12562) - 

Reviewer 2: No, let's remove it. We've mentioned it in 2.40.1 RN

> - [PMM-12576](https://jira.percona.com/browse/PMM-12576) - 

Reviewer 2: let's remove

> - [PMM-12630](https://jira.percona.com/browse/PMM-12630) - 

Reviewer 2: let's remove

> - [PMM-12630](https://jira.percona.com/browse/PMM-12630) - 

Author: Ok.

> - [PMM-12660](https://jira.percona.com/browse/PMM-12660) - 

Reviewer 2: looks ok to me

# Percona Monitoring and Management 2.41.0

> | **Release date** | Nov 25, 2023                                                                                    |

Reviewer: ```suggestion
| **Release date** | Dec 6, 2023                                                                                    |
```
